### ISSUE TITLE
KubeCluster Docs Typo

### ISSUE THREAD

**Issue by chriscardillo** [dask/dask-kubernetes#665]
**Describe the issue**: [Latest docs for using the `KubeCluster` class](https://kubernetes.dask.org/en/latest/#kubecluster) uses a `name` that is not a "a lowercase RFC 1123 subdomain", which cases a 403 error when ran in a Kubernetes cluster.

**Minimal Complete Verifiable Example**: Run the [latest docs](https://kubernetes.dask.org/en/latest/#kubecluster) example:

```python3
from dask_kubernetes.operator import KubeCluster
cluster = KubeCluster(name="my_dask_cluster", image='ghcr.io/dask/dask:latest')
```

**Anything else we need to know?**: `my-dask-cluster` will work as an alternative.

**Environment**:

- Dask version: `dask==2023.3.0`, `dask-kubernetes==2023.1.1`
- Python version: `3.9.14`
- Operating System: `Debian GNU/Linux 11 (bullseye)`
- Install method (conda, pip, source): `pip`
- Kubernetes version: `1.25.3`

**Comment by jacobtomlinson** [comment]
Thanks for raising this. I see two issues here:
- Our docs have an invalid name
- We are not doing any input validation on the name
It would be good to address both of those problems.

**Comment by jacobtomlinson** [comment]
Closed the immediate bug in #681 and opened #682 to track future work around input sanitising.
